sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/Config/SignalROptions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Azure.SignalR;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    /// <summary>
+    /// Configuration options for SignalR extensions.
+    /// </summary>
+    /// <remarks>
+    /// Most properties here can be configured in <see cref="IConfiguration"/>, except JSON serialization.
+    /// </remarks>
+    public class SignalROptions
+    {
+        /// <summary>
+        /// Gets or sets the service endpoints.
+        /// </summary>
+        public ServiceEndpoint[] ServiceEndpoints { get; set; }
+
+        /// <summary>
+        /// Gets or sets the service transport type.
+        /// </summary>
+        public ServiceTransportType? ServiceTransportType { get; set; }
+
+        /// <summary>
+        /// Use Newtonsoft.Json as the JSON serialization library.
+        /// </summary>
+        /// <param name="configure">Configure the Newtonsoft Json service hub protocol options.</param>
+        public void UseNewtonsoftJson(Action<NewtonsoftServiceHubProtocolOptions> configure)
+        {
```

Comment:
For reference, here is how we exposed Newtonsoft settings in Service Bus - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/api/Microsoft.Azure.WebJobs.Extensions.ServiceBus.netstandard2.0.cs#L78

Reply:
SignalR also supports `System.Text.Json`. We might need to allow users to choose the library and the serializer settings at the same time. For example, Asp.Net Core SignalR uses [`AddNewtonsoftJson` method](https://source.dot.net/#Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson/NewtonsoftJsonProtocolDependencyInjectionExtensions.cs,1a63c8f2c2872010,references) to set protocol and the protocol settings.

Reply:
How about exposing one enum field to hold protocol name, "NewtonsoftJson" or "SystemTextJson", and another `object` field to hold the serializer settings?

Reply:
Don't we need to configure the NewtonSoftServiceHubProtocolOptions?
I'm not opposed to having UseNewtonsoftJson - I was just showing how we did it in SB. You raise a fair point that if we need to eventually support STJ, this may be a better pattern. Still, we should try to get this reviewed by the .NET architects and the functions folks.